### ISSUE TITLE
metakernel won't work on unix systems without bash

### ISSUE THREAD

**Issue by bsvh** [Calysto/metakernel#90]
I tried using the octave_kernel on an OpenBSD system that does not have bash installed, but it fails because of Metakernel. I was able to fix this easily by changing to `def bash(command="sh", prompt_regex=re.compile('[$#]'))` of replwrap.py, and `pexpect.which('sh')` in shell_magics.py, since it doesn't seem like anything you are using bash for in this case is bash-specific. However, I'm not sure whether it affects the MetaKernel Bash or not.

**Comment by blink1073** [comment]
Does making this change address your issue?

```diff
--- a/metakernel/magics/shell_magic.py
+++ b/metakernel/magics/shell_magic.py
@@ -65,7 +65,8 @@ class ShellMagic(Magic):
                 if os.name == 'nt':
                     self.cmd = 'cmd'
                 else:
-                    raise(e)
+                    pexpect.which('sh')
+                    self.cmd = 'sh'
             else:
                 self.cmd = 'bash'

**Comment by blink1073** [comment]
If so, can you update your PR?

**Comment by bsvh** [comment]
That fixed it, and I updated my PR. Thanks.

**Comment by bsvh** [comment]
Scratch that, I wasn't working on the live codebase. This isn't working because the bash function is still trying to use bash. I think this can be fixed by using `self.repl = bash(command=self.cmd)` afterwords. I'll go test it.

**Comment by bsvh** [comment]
The problem was that `pexpect.which('bash')` doesn't raise an exception, it just returns an empty string. My updated pull request just checks for an empty string.

**Comment by blink1073** [comment]
Fixed by #89.
